host/acquire_shard_integration_test.go

```diff
@@ -0,0 +1,120 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package host
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+)
+
+type acquireShardIntegrationSuite struct {
+	IntegrationBase
+	logRecorder *logRecorder
+	logs        chan string
+}
+
+func TestAcquireShardIntegrationSuite(t *testing.T) {
+	s := new(acquireShardIntegrationSuite)
+	suite.Run(t, s)
+}
+
+func (s *acquireShardIntegrationSuite) SetupSuite() {
+	// need to buffer the logs channel so that the test doesn't block
+	s.logs = make(chan string, 1000)
+	s.logRecorder = NewLogRecorder(s.logs)
+	s.Logger = s.logRecorder
+	s.setupSuite("testdata/acquire_shard_ownership_lost_error.yaml")
+}
+
+func (s *acquireShardIntegrationSuite) TearDownSuite() {
+	// give other services a chance to start up before shutting down
+	time.Sleep(time.Second * 5)
+	s.tearDownSuite()
+}
+
+func (s *acquireShardIntegrationSuite) TestAcquireShard_OwnershipLostErrorDoesNotRetry() {
```

Comment:
We need one more test case to verify when the injected error is not ShardOwnershipLost, the shard won't be closed.

Reply:
Added.